pkg/registry/batch/rest/storage_batch.go

```diff
@@ -36,6 +37,10 @@ func (p RESTStorageProvider) NewRESTStorage(apiResourceConfigSource genericapise
 	if apiResourceConfigSource.AnyResourcesForVersionEnabled(batchapiv2alpha1.SchemeGroupVersion) {
 		apiGroupInfo.VersionedResourcesStorageMap[batchapiv2alpha1.SchemeGroupVersion.Version] = p.v2alpha1Storage(apiResourceConfigSource, restOptionsGetter)
 		apiGroupInfo.GroupMeta.GroupVersion = batchapiv2alpha1.SchemeGroupVersion
+		apiGroupInfo.SubresourceGroupVersionKind = map[string]unversioned.GroupVersionKind{
```

Comment:
This tells the apiserver to create the codec to serialize and deserialize from ScheduledJob in preference to CronJob for that endpoint (otherwise it uses the group version's preferred kind, which is CronJob)